kura/org.eclipse.kura.api/src/main/java/org/eclipse/kura/cloudconnection/factory/CloudConnectionManagerFactory.java

```diff
@@ -0,0 +1,112 @@
+/*******************************************************************************
+ * Copyright (c) 2018 Eurotech and/or its affiliates
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Eurotech
+ *******************************************************************************/
+package org.eclipse.kura.cloudconnection.factory;
+
+import java.util.List;
+import java.util.Set;
+
+import org.eclipse.kura.KuraException;
+import org.eclipse.kura.cloudconnection.CloudConnectionManager;
+import org.osgi.annotation.versioning.ProviderType;
+
+/**
+ * A {@link CloudConnectionManagerFactory} represents an OSGi Declarative Service Component
+ * that registers {@link CloudConnectionManager}s in the framework.
+ * The Component creates multiple component instances upon reception of a configuration
+ * created through the Configuration Service.
+ *
+ * It provides a {@link CloudConnectionManager} implementation that can be used to connect to a specific Cloud platform.
+ *
+ * Typically, each {@link CloudConnectionManager} created by a {@link CloudConnectionManagerFactory}
+ * establishes and manages its own connection, for example an Mqtt connection.
+ *
+ * Multiple CloudServiceFactory services can be registered in the framework to support multiple simultaneous
+ * connections to different Cloud platforms.
+ *
+ * A CloudServiceFactory manages the construction of a CloudService and the services it depends on.
+ *
+ * @noimplement This interface is not intended to be implemented by clients.
+ * @since 2.0
+ */
+@ProviderType
+public interface CloudConnectionManagerFactory {
+
+    /**
+     * The name of the property set in a @{link CloudConnectionManager} configuration created
+     * through {@link #createConfiguration}.
+     * The property is set in the {@link CloudConnectionManager} instance to relate it with the Factory that generated
+     * the whole cloud stack.
+     */
+    public static final String KURA_CLOUD_CONNECTION_SERVICE_FACTORY_PID = "kura.cloud.connection.service.factory.pid";
```

Comment:
Just to be concise, you could remove the access modifier from all the methods and constants from this interface.